ompi/mca/coll/base/coll_base_dynamic_file.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2020      Bull SAS. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "mpi.h"
+#include "ompi/mca/mca.h"
+#include "base.h"
+
+/* need to include our own topo prototypes so we can malloc data on the comm correctly */
+#include "coll_base_topo.h"
+
+/* need file reading function */
+#include "coll_base_util.h"
+
+/* also need the dynamic rule structures */
+#include "coll_base_dynamic_rules.h"
+
+/* and our own prototypes */
+#include "coll_base_dynamic_file.h"
+
+static int fileline=0; /* used for verbose error messages */
+
+#define getnext(fptr, pval)        ompi_coll_base_file_getnext_long(fptr, &fileline, pval)
+
+/*
+ * Reads a rule file called fname
+ * Builds the algorithm rule table for a max of n_collectives
+ *
+ * If an error occurs it removes rule table and then exits with a very verbose
+ * error message (this stops the user using a half baked rule table
+ *
+ * Returns the number of actual collectives that a rule exists for
+ * (note 0 is NOT an error)
+ *
+ */
+
+int ompi_coll_base_read_rules_config_file (char *fname, int format_version, ompi_coll_base_alg_rule_t** rules, int n_collectives)
```

Comment:
This seems to be major new functionality.

Can you provide some documentation for what it is, how it works, how users use it, why users should use it, ...etc.?

Can you also provide some test cases in https://github.com/open-mpi/ompi-tests-public?

Reply:
This is a renaming (and moving) of the tuned capability (aka. ompi_coll_tuned_read_rules_config_file).